Mathlib/Combinatorics/Enumerative/DyckWord.lean

```diff
@@ -297,22 +317,19 @@ lemma firstReturn_nest : p.nest.firstReturn = p.toList.length + 1 := by
   · simp_rw [length_range, u, length_append, length_cons]
     exact Nat.lt_add_one _
 
-lemma firstReturn_nest_add : (p.nest + q).firstReturn = p.toList.length + 1 := by simp
```

Comment:
Where is this lemma gone? Was it supposed to be an auxiliary lemma and you don't need it anymore?

Reply:
It was originally needed in the proofs of simp lemmas `insidePart_nest_add` and `outsidePart_nest_add`, and those were its only two uses. When I was cleaning up this PR after merging master just after #15864 landed, I needed to split `insidePart_nest_add` and `outsidePart_nest_add` into two more elementary simp lemmas each to prove `DyckWord.le_add_self`, so I did that. Then I realised that I no longer needed to explicitly state `insidePart_nest_add`/`outsidePart_nest_add` in the only place where _they_ were used, in the proof of `treeEquiv_right_inv`, because the four simp lemmas replacing them are enough for `simp` to close the goal. So I could dispense with `insidePart_nest_add`, `outsidePart_nest_add` – and transitively `firstReturn_nest_add`.